src/progress/pct/v11/dynrun.p

```diff
@@ -0,0 +1,136 @@
+USING Progress.Json.ObjectModel.JsonArray.
+USING Progress.Json.ObjectModel.JsonObject.
+USING Progress.Json.ObjectModel.ObjectModelParser.
+
+ROUTINE-LEVEL ON ERROR UNDO, THROW.
+
+DEFINE NEW SHARED VARIABLE pctVerbose AS LOGICAL NO-UNDO.
+DEFINE VARIABLE noErrorOnQuit AS LOGICAL NO-UNDO.
+
+DEFINE VARIABLE i AS INTEGER NO-UNDO INITIAL ?.
+
+DEFINE TEMP-TABLE ttParams NO-UNDO
+  FIELD key AS CHARACTER
+  FIELD val AS CHARACTER.
+
+FUNCTION getParameter RETURNS CHARACTER (k AS CHARACTER).
+  FIND ttParams WHERE ttParams.key EQ k NO-LOCK NO-ERROR.
+  RETURN (IF AVAILABLE ttParams THEN ttParams.val ELSE ?).
+END FUNCTION.
+
+DEFINE VARIABLE jsonParser AS CLASS ObjectModelParser NO-UNDO.
+DEFINE VARIABLE configJson AS CLASS JsonObject NO-UNDO.
+
+DEFINE VARIABLE ppEntries AS CLASS JsonArray NO-UNDO.
+DEFINE VARIABLE dbEntries AS CLASS JsonArray NO-UNDO.
+DEFINE VARIABLE dbEntry   AS CLASS JsonObject NO-UNDO.
+DEFINE VARIABLE prmEntries AS CLASS JsonArray NO-UNDO.
+DEFINE VARIABLE prmEntry   AS CLASS JsonObject NO-UNDO.
+DEFINE VARIABLE outprmEntries AS CLASS JsonArray NO-UNDO.
+DEFINE VARIABLE zz AS INTEGER     NO-UNDO.
+DEFINE VARIABLE xx AS INTEGER     NO-UNDO.
+DEFINE VARIABLE out1 AS CHARACTER   NO-UNDO.
+DEFINE VARIABLE out2 AS CHARACTER   NO-UNDO.
+
+ASSIGN jsonParser = NEW ObjectModelParser().
+ASSIGN configJson = CAST(jsonParser:ParseFile(SESSION:PARAMETER), JsonObject).
+
+ASSIGN pctVerbose = configJson:getLogical("verbose").
+
+// DB connections + aliases
+ASSIGN dbEntries = configJson:GetJsonArray("databases").
+DO zz = 1 TO dbEntries:Length:
+  ASSIGN dbEntry = dbEntries:GetJsonObject(zz).
+  IF pctVerbose THEN MESSAGE "Connecting to '" + dbEntry:getCharacter("connect") + "'".
+  CONNECT VALUE(dbEntry:getCharacter("connect")) NO-ERROR.
+  IF ERROR-STATUS:ERROR THEN DO:
+    MESSAGE "Unable to connect to '" + dbEntry:getCharacter("connect") + "'".
+    DO i = 1 TO ERROR-STATUS:NUM-MESSAGES:
+      MESSAGE ERROR-STATUS:GET-MESSAGE(i).
+    END.
+    RUN returnValue(14).
+    QUIT.
+  END.
+  DO xx = 1 TO dbEntry:getJsonArray("aliases"):Length:
+    IF pctVerbose THEN MESSAGE SUBSTITUTE("Creating alias &1 for database #&2 &3", dbEntry:getJsonArray("aliases"):getCharacter(xx), zz, LDBNAME(zz)).
+    CREATE ALIAS VALUE(dbEntry:getJsonArray("aliases"):getCharacter(xx)) FOR DATABASE VALUE(LDBNAME(zz)) NO-ERROR.
+    IF ERROR-STATUS:ERROR THEN DO:
+      MESSAGE SUBSTITUTE("Unable to create alias '&1' for database '&2'", dbEntry:getJsonArray("aliases"):getCharacter(xx),  LDBNAME(zz) ).
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Line is 139 characters long, exceeding maximum value of 120 [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.LinesTooLong)